design/sequence/scheduler-low.puml

```diff
@@ -0,0 +1,86 @@
+@startuml
+
+box "Central Server" #Lightblue
+    participant ":Database"
+    participant ":Scheduler"
+    participant "job:Job"
+    participant ":Dispatcher"
+end box
+
+
+":Database" -> ":Scheduler" ++: reschedule()
+":Scheduler" -> ":Database" ++: get_changed_job()
+":Database" --> ":Scheduler" --: job: Job
+activate "job:Job"
+
+
+break job.get_status()==JobStatus.KILLED||\n job.get_status()==JobStatus.FINISHED
+    ":Scheduler" -> "job:Job" ++:check_was_running()
+    "job:Job" --> ":Scheduler" --: was_running: bool
+
+    alt was_running
+        alt job.get_status()==JobStatus.KILLED
+            ":Scheduler" -> ":Dispatcher" ++: kill_job(job)
+            ":Dispatcher" --> ":Scheduler" --
+        end
+        ":Scheduler" -> ":Database" ++: get_next_job(job.wc_id)
+        ":Database" --> ":Scheduler" --: new_job: Job
+        ":Scheduler" -> ":Dispatcher" ++ : dispatch_job(new_job)
+        ":Dispatcher" --> ":Scheduler" --:Message
+        ":Scheduler" -> ":Database"++: set_job_status(new_job, JobStatus.RUNNING)
+        ":Database" --> ":Scheduler" --
+    end
+end
+
+":Scheduler" -> ":Scheduler" ++: find_wcs(job)
+":Scheduler" -> ":Database" ++: get_wc_list()
+":Database" --> ":Scheduler" --: wc_list:list[str]
+":Scheduler" --> ":Scheduler" --: suitable_wcs:list[str]
```

Comment:
But several work machines can run a job (since they can have the same hardware specifications). We store those potential work machines in the Job instance. Once a work machine finishes a job, the Scheduler iterates over the queued jobs and finds the Job with the highest effective priority that could run on that work machine.

Reply:
I think we'll do it the other way, going through the list of jobs, we try to find a work machine to put it in.

Reply:
But several work machines can run a job (since they can might have the same hardware specifications). We store those potential work machines in the Job instance. Once a work machine finishes a job, the Scheduler iterates over the queued jobs and finds the Job with the highest effective priority that could run on that work machine.

Reply:
@ammen99 Agreed (I deleted that comment, didn't think you read it already).

Reply:
This should be determined on-the-fly at the beginning of the scheduling algorithm.